SU2_CFD/src/numerics/heat.cpp

```diff
@@ -71,16 +56,27 @@ void CCentSca_Heat::ComputeResidual(su2double *val_residual, su2double **val_Jac
 
   /*--- Projected velocities at the current edge ---*/
 
-  ProjVelocity = 0.0; ProjVelocity_i = 0.0; ProjVelocity_j = 0.0;
-  for (iDim = 0; iDim < nDim; iDim++) {
-    ProjVelocity_i += V_i[iDim+1]*Normal[iDim];
-    ProjVelocity_j += V_j[iDim+1]*Normal[iDim];
+  su2double ProjVelocity_i = 0.0;
+  su2double ProjVelocity_j = 0.0;
+
+  if (dynamic_grid) {
+    for (unsigned short iDim = 0; iDim < nDim; iDim++) {
+      su2double Velocity_i = V_i[iDim+1] - GridVel_i[iDim];
+      su2double Velocity_j = V_j[iDim+1] - GridVel_j[iDim];
+      ProjVelocity_i += Velocity_i*Normal[iDim];
+      ProjVelocity_j += Velocity_j*Normal[iDim];
+    }
+  }
+  else {
+    for (unsigned short iDim = 0; iDim < nDim; iDim++) {
+      ProjVelocity_i += V_i[iDim+1]*Normal[iDim];
+      ProjVelocity_j += V_j[iDim+1]*Normal[iDim];
+    }
   }
-  Area = GeometryToolbox::Norm(nDim, Normal);
 
   /*--- Computing the second order centered scheme part ---*/
 
-  ProjVelocity = 0.5*(ProjVelocity_i+ProjVelocity_j);
+  su2double ProjVelocity = 0.5*(ProjVelocity_i+ProjVelocity_j);
```

Comment:
```suggestion
  const su2double ProjVelocity = 0.5*(ProjVelocity_i+ProjVelocity_j);
```

Reply:
any reason to not const all of those? Does the compiler do the necessary optimization anyway because the scope is so narrow?

Reply:
I don't think compilers look for const to optimize, because you have things like const_cast as part of legal c++.
I was lazy... You can go ahead and constify if you want but the method is small and safe.

Reply:
Ok you already went ahead yourself 👍